packages/cli/src/commands/site/deploy/index.ts

```diff
@@ -14,54 +14,126 @@
+    CommonSiteArgs,
+    SiteDeployArgs
+  > = {
+    command: "deploy",
+    describe: "Deploy an uploaded version",
+    builder: (argv) => {
+      return argv
+        .options({
+          directory: {
+            type: "string",
+            description: "Directory to deploy",
+            ...directory
+              ? { default: directory }
+              : { demandOption: true },
+          },
+          uploadOnly: {
+            type: "boolean",
+            description: "Upload the directory but do not deploy it",
+            default: false,
+          },
+          version: {
+            type: "string",
+            description: "Version to deploy",
+            ...autoVersion == null
+              ? { conflicts: "autoVersion" }
+              : {},
+          },
+          autoVersion: {
+            type: "string",
+            description:
+              "Enables autoversioning. Can be set to 'git-describe' to use git describe to determine the version.",
+            ...(autoVersion != null)
+              ? { default: autoVersion.type }
+              : { conflicts: "version" },
+          },
+          gitTagPrefix: {
+            type: "string",
+            description:
+              "Prefix to match git tags against when --autoVersion=git-describe is used. If not provided, a default prefix 'v' is used.",
+            ...gitTagPrefix
+              ? { default: gitTagPrefix }
+              : {},
+          },
+        }).group(
+          ["autoVersion", "gitTagPrefix"],
+          "Autoversion Arguments",
+        )
+        .group(
+          ["version", "directory", "uploadOnly"],
+          "Common Arguments",
+        )
+        .check((argv) => {
+          // This is required because we can't use demandOption with conflicts. conflicts protects us against the case where both are provided.
+          // So this case is for when nothing is provided.
+          if (
+            autoVersion == null && argv.autoVersion == null
+            && argv.version == null
+          ) {
+            throw new Error(
+              "One of --version or --autoVersion must be specified",
+            );
+          }
 
-export default command;
+          if (
+            (autoVersion?.type != "git-describe"
+              || argv.autoVersion != "git-describe")
+            && argv.gitTagPrefix != null
+          ) {
+            throw new Error(
+              `--gitTagPrefix is only supported when --autoVersion=git-describe`,
+            );
+          }
+
+          if (autoVersion != null && argv.autoVersion !== autoVersion.type) {
+            consola.debug(
+              `Overriding "autoVersion" from config file with ${argv.autoVersion}`,
+            );
+            if (argv.autoVersion != "git-describe") {
+              throw new Error(
+                `Only 'git-describe' is supported for autoVersion`,
+              );
+            }
+          }
+
+          if (directory != null && argv.directory !== directory) {
```

Comment:
As a follow up I think these logging side effects also don't make sense to be part of "check" as they aren't actually checking for any violations